packages/graph/hash_graph/bench/benches/read_scaling/knowledge/complete/entity.rs

```diff
@@ -0,0 +1,276 @@
+            store
+                .get_entity(&StructuralQuery {
+                    filter: Filter::for_entity_by_entity_id(entity_edition_id.base_id()),
+                    graph_resolve_depths,
+                })
+                .await
+                .expect("failed to read entity from store");
+        },
+        SmallInput,
+    );
+}
+
+#[criterion]
+fn bench_scaling_read_entity_zero_depths(c: &mut Criterion) {
+    let mut group = c.benchmark_group("scaling_read_entity_complete_zero_depth");
+
+    group.sample_size(10);
+    group.sampling_mode(SamplingMode::Flat);
+
+    // We use a hard-coded UUID to keep it consistent across tests so that we can use it as a
+    // parameter argument to criterion and get comparison analysis
+    let account_id =
+        AccountId::new(Uuid::from_str("bf5a9ef5-dc3b-43cf-a291-6210c0321eba").unwrap());
+
+    for size in [1, 5, 10, 25, 50] {
+        // TODO: reuse the database if it already exists like we do for representative_read
+        let (runtime, mut store_wrapper) = setup(DB_NAME, true, true);
+
+        let DatastoreEntitiesMetadata {
+            entity_metadata_list,
+            ..
+        } = runtime.block_on(seed_db(account_id, &mut store_wrapper, size));
+        let store = &store_wrapper.store;
+
+        group.bench_with_input(
+            BenchmarkId::new(
+                "get_entity_by_id",
+                format!("Account ID: `{account_id}`, Number Of Entities: `{size}`"),
+            ),
+            &(account_id, entity_metadata_list),
+            |b, (_account_id, entity_metadata_list)| {
+                bench_get_entity_by_id(
+                    b,
+                    &runtime,
+                    store,
+                    entity_metadata_list,
+                    GraphResolveDepths {
+                        inherits_from: Default::default(),
+                        constrains_values_on: Default::default(),
+                        constrains_properties_on: Default::default(),
+                        constrains_links_on: Default::default(),
+                        constrains_link_destinations_on: Default::default(),
+                        is_of_type: Default::default(),
+                        has_left_entity: Default::default(),
+                        has_right_entity: Default::default(),
+                    },
+                )
+            },
+        );
+    }
+}
+
+#[criterion]
+fn bench_scaling_read_entity_one_depth(c: &mut Criterion) {
+    let mut group = c.benchmark_group("scaling_read_entity_complete_one_depth");
+
+    group.sample_size(10);
+    group.sampling_mode(SamplingMode::Flat);
+
+    // We use a hard-coded UUID to keep it consistent across tests so that we can use it as a
+    // parameter argument to criterion and get comparison analysis
+    let account_id =
+        AccountId::new(Uuid::from_str("bf5a9ef5-dc3b-43cf-a291-6210c0321eba").unwrap());
+
+    for size in [1, 5, 10, 25, 50] {
```

Comment:
I think we shouldn't use `50` here as the benchmark will run quite a while (2500 links).
```suggestion
    for size in [1, 5, 10, 25 /* , 50 */] {
```

Reply:
It takes about 10 minutes for that specific benchmark, which is bad and I'm already slightly concerned about how long our benchmarks take to run. However, for scenarios where we don't particularly care I think we can just disable the benchmark, but in most scenarios I think this case is quite important as it's hard to see how poor the performance change is as the numbers increase (as in 1 -> 5 -> 10 doesn't paint a good picture, so we kind of need 25 -> 50)

```
Finished seeding database entity_scale with 5 entities and 25 link entities after 271.603ms
Benchmarking scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #2: Warming up for 3.0000 s
Warning: Unable to complete 10 samples in 5.0s. You may wish to increase target time to 8.3s.
scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #2
                        time:   [725.90 ms 791.88 ms 853.66 ms]
                        change: [-23.084% -13.450% -3.2853%] (p = 0.04 < 0.05)
                        Performance has improved.
Seeding database: entity_scale
Finished seeding database entity_scale with 10 entities and 100 link entities after 335.602ms
Benchmarking scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #3: Warming up for 3.0000 s
Warning: Unable to complete 10 samples in 5.0s. You may wish to increase target time to 33.4s.
scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #3
                        time:   [2.5267 s 2.5890 s 2.6570 s]
                        change: [-20.533% -15.123% -10.038%] (p = 0.00 < 0.05)
                        Performance has improved.
Seeding database: entity_scale
Finished seeding database entity_scale with 25 entities and 625 link entities after 569.674ms
Benchmarking scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #4: Warming up for 3.0000 s
Warning: Unable to complete 10 samples in 5.0s. You may wish to increase target time to 146.6s.
scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #4
                        time:   [14.913 s 15.191 s 15.518 s]
                        change: [-17.158% -14.440% -11.811%] (p = 0.00 < 0.05)
                        Performance has improved.
Seeding database: entity_scale
Finished seeding database entity_scale with 50 entities and 2500 link entities after 626.593ms
Benchmarking scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #5: Warming up for 3.0000 s
Warning: Unable to complete 10 samples in 5.0s. You may wish to increase target time to 571.1s.
scaling_read_entity_complete_one_depth/get_entity_by_id/Account ID: `bf5a9ef5-dc3b-43cf-a291-6210c03... #5
                        time:   [57.350 s 57.491 s 57.650 s]
                        change: [-21.605% -19.122% -16.836%] (p = 0.00 < 0.05)
                        Performance has improved.
```

Notice 15s to 57s, I think this is one of the most important metrics we should be targeting to bring down as we optimise further.

Reply:
Good call, let's leave it in. 🙂